### PR TITLE
SW-5168 Project edit form fixes

### PR DESCRIPTION
**Pull request by nickgraz**
- Change number field to number type `Textfield` in accelerator project edit. 
- Add '(ha)' to '* reforestable land' fields
- Fix race condition for project meta date rendering.

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#2478** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2478?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/ProjectField/Meta.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ const ProjectFieldMeta = ({ date, dateLabel, userId, userName, userLabel }: Proj
               marginBottom={theme.spacing(1)}
               component={'span'}
             >
-              {getDateDisplayValue(date || '')}
+              {date ? getDateDisplayValue(date) : ''}
```
Comment (karthikbtf) [inline src/components/ProjectField/Meta.tsx:43]: I suppose this fixes https://terraformation.atlassian.net/browse/SW-5173

**Review by karthikbtf — APPROVED** [review]
LGTM 🎉

File: src/components/ProjectField/Meta.tsx (hunk 1 above)
Comment (nickgraz) [reply src/components/ProjectField/Meta.tsx:43]: I believe so, it was a race condition so somewhat hard to reproduce.

**Comment by nickgraz** [comment]
### Merge activity

* **Apr 4, 1:23 PM EDT**: @nickgraz merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2478).

**Pull request merged**